.github/workflows/main.yml

```diff
@@ -0,0 +1,20 @@
+on: [push, pull_request]
+
+name: Lint with awesome-lint
+
+jobs:
+  awesome-lint-workflow:
+    runs-on: ubuntu-latest
+    name: Linting with awesome-lint
+    steps:
+      - name: checkout repo
+        id: checkout
+        uses: actions/checkout@v2
+      - name: awesome-lint-action
+        uses: squadcastHQ/awesome-lint-action@master
```

Comment:
I don't really see why you need a custom workflow for this. You could just run `$ awesome-lint` as a command.